journals/week5.md

```diff
@@ -0,0 +1,186 @@
+
+### Adjusting the Resolution of metrics
+
+One problem with the current stack is timing. I had setup the timer watcher in the plugin to send metrics once per second, then brubeck was aggregating and sending metrics over a 10s interval, and on the backend graphite *seemed* to be aggregating and plotting data once per minute or so. Given that some of the job states happen way faster than 1s or 10s or 1m, I tried to see if I could tweak some of the intervals to see a more finite resolution of each state and hopefully a more accurate picture of what was going on without impacting performance too much. One thing to note is going forward I started using flux instances of size 8 to see how that would effect the performance. The first thing I did was try adjusting the frequency at which the timer watcher was called to 0.1s and then 0.025s and running 20,000 jobs under each in two batches of 10,000. The 0.1s timer watcher ran first from 20:13-20:17 and the 0.025s watcher ran second from 20:18-20:21.
+
+**Inactive vs Sched**
+
+![job states sched vs inactive](../img/gr_flux_js_is_1.png)
+
+**Cleanup vs Run**
+
+![job states cleanup vs run](../img/gr_flux_js_cr_1.png)
+
+The shorter interval for the watcher had a very minimal performance effect on the jobs themselves, and it did seem to give a little bit more precision wrt to the `sched` and `inactive` states since they are closer to the true starting numbers of 0 and 10,000. It's also interesting to note that this was run on an instance of size 8 so with my 12 core CPU the instance had access to 96 virtual cores, but it never got more than 70 cores at use at once (at least that was reported). It was using on average only about half of the available cores at a time. Then I decided to pick 0.1s as the interval for the timer watcher due mostly to the type being used for the metrics since each time it's called the previous value is just overwritten in the aggregator anyways. Next, I tried adjusting the flush interval within brubeck to 10s and then 1s and running 10,000 jobs in a flux instance for each interval.
+
+**Inactive vs Sched**
```

Comment:
Once jobs enter the INACTIVE state, they should never leave it. Any sense on why the count of INACTIVE jobs drops in these graphs? (I'm probably just missing something simple sorry.)